tutor/src/screens/gradebook/weights-ux.js

```diff
@@ -0,0 +1,203 @@
+import { observable, computed, action } from 'mobx';
+import { sum, toArray, flow, inRange, keys, isEqual, pick, invert, mapValues, isNaN, partial } from 'lodash';
+
+const CELL_AVERAGES_SINGLE_WIDTH = 80;
+
+const DEFAULTS = {
+  homework_scores: 100,
+  homework_progress: 0,
+  reading_scores: 0,
+  reading_progress: 0,
+};
+
+const RECOMMENDED = {
+  homework_scores: 50,
+  homework_progress: 0,
+  reading_scores: 0,
+  reading_progress: 50,
+};
+
+const CW = {
+  homework_score_weight: 'homework_scores',
+  homework_progress_weight: 'homework_progress',
+  reading_score_weight: 'reading_scores',
+  reading_progress_weight: 'reading_progress',
+};
+
+const WC = invert(CW);
+
+const PERCENT_KEYS = keys(DEFAULTS);
+const WEIGHT_KEYS = keys(CW);
+
+const MIN = 0;
+const MAX = 100;
+const RANGE = MAX - MIN;
+
+const weightToPercent = (weight) => ((weight || MIN) * RANGE );
+const percentToWeight = (percent) => ((percent || MIN) / RANGE );
+
+const weightsToPercents = (weights) => mapValues(WC, (c) => weightToPercent(weights[c]));
+const percentsToWeights = (percents) => mapValues(CW, (w) => percentToWeight(percents[w]));
+
+const stringToInt = (string) => {
+  const int = parseInt(string);
+
+  if (isNaN(int)) {
+    return 0;
+  }
+
+  return int;
+};
+
```

Comment:
there's a `string` helper at `helpers/string`. How about adding this to that file, I seems like it'd be helpful elsewhere.

Reply:
:+1: